nibabel/info.py

```diff
@@ -19,6 +23,51 @@
                               _version_micro,
                               _version_extra)
 
+
+_VERSION_RE = re.compile(r'([0-9.]*\d+)(.*)')
+
+
+def _parse_version(version_str):
+    """ Parse version string `version_str` in our format
+    """
+    match = re.match('([0-9.]*\d)(.*)', version_str)
+    if match is None:
+        raise ValueError('Invalid version ' + version_str)
+    return match.groups()
+
+
+def _cmp(a, b):
+    """ Implementation of ``cmp`` for Python 3
+    """
+    return (a > b) - (a < b)
+
+
+def cmp_pkg_version(version_str, pkg_version_str=__version__):
+    """ Compare `version_str` to current package version
+
+    Parameters
+    ----------
+    version_str : str
+        Version string to compare to current package version
+    pkg_version_str : str, optional
+        Version of our package.  Optional, set fom ``__version__`` by default.
+
+    Returns
+    -------
+    version_cmp : int
+        1 if `version_str` higher version than `pkg_version`, 0 if same, -1 if
+        lower.
+    """
+    version, extra = _parse_version(version_str)
+    pkg_version, pkg_extra = _parse_version(pkg_version_str)
+    if version != pkg_version:
+        return _cmp(StrictVersion(version), StrictVersion(pkg_version))
+    return (0 if extra == pkg_extra
+            else 1 if extra == ''
+            else -1 if pkg_extra == ''
+            else _cmp(extra, pkg_extra))
```

Comment:
Looks like you're planning for extras like `'a'` and `'b'`? As long as you don't do `'-rc1'` or anything (where `'3.0-rc1'` should be <  `'3.0'`) this seems fine.